spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -361,4 +403,90 @@ static PathFilter forSpecs(Map<Integer, PartitionSpec> specs) {
       return partitionNames.isEmpty() ? HiddenPathFilter.get() : new PartitionAwareHiddenPathFilter(partitionNames);
     }
   }
+
+  private static Dataset<Row> sanitizeAndSplitPaths(Dataset<Row> filesDF) {
```

Comment:
This class uses a slightly different style for defining Spark transforms (see `listDirsRecursively` above).

What about something like this?

```
  private static MapPartitionsFunction<Row, Row> toFileMetadata(Map<String, String> equalSchemes,
                                                                Map<String, String> equalAuthorities) {
    return rows -> Iterators.transform(rows, row -> {
      String filePathAsString = row.getString(0);
      URI uri = new Path(filePathAsString).toUri();

      String scheme = uri.getScheme();
      String normalizedScheme = equalSchemes.getOrDefault(scheme, scheme);

      String authority = uri.getAuthority();
      String normalizedAuthority = equalAuthorities.getOrDefault(authority, authority);

      return RowFactory.create(normalizedScheme, normalizedAuthority, uri.getPath(), filePathAsString);
    });
  }
```

Then we can call it like this:

```
Dataset<Row> validFileMetadataDF = validFileDF
    .mapPartitions(toFileMetadata(equalSchemes, equalAuthorities), FILE_METADATA_ENCODER)
    .as("valid");
```